src/main/java/uk/gov/ch/oauth/session/SessionUtils.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.ch.oauth.session;
+
+import java.util.Map;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.handler.SessionHandler;
+import uk.gov.companieshouse.session.store.Store;
+
+
+/**
+ * This is a wrapper component for {@link uk.gov.companieshouse.session.SessionFactory} that allows
+ * for mocking return results, rather than the implemented static methods.
+ *
+ * @see uk.gov.companieshouse.session.SessionFactory#getSessionByCookieId(String)
+ * @see uk.gov.companieshouse.session.SessionFactory#getSessionByCookieId(Store, String)
+ */
+@Component
+public class SessionUtils {
+
+    public Map<String, Object> getSessionDataFromContext() {
+        return SessionHandler.getSessionDataFromContext();
+    }
+
+    Session getSessionByCookieId(final String cookieId) {
+        return uk.gov.companieshouse.session.SessionFactory.getSessionByCookieId(cookieId);
+    }
+
+    public Session createSession() {
+        return getSessionByCookieId(null);
+    }
+}
```

Comment:
What is the difference between this and the SessionFactory we have in Docs? Can we look to combine the classes.

Reply:
uk.gov.ch.developer.docs.session.SessionFactory

Reply:
The difference is that this is part of the oauth package which will hopefully make it easier to extract the library.